backend/src/main/java/com/example/backend/Controllers/AdminDashboard/FlightManagementController.java

```diff
@@ -23,4 +26,16 @@ public ResponseEntity<String> deleteFlight(@PathVariable int id) {
         flightManagmentService.deleteFlight(id);
         return ResponseEntity.ok("The flight has been deleted successfully.");
     }
+
+    @DeleteMapping("/flights/cancel/{id}")
+    public ResponseEntity<String> cancelFlight(@PathVariable int id) {
+        flightManagmentService.cancelFlight(id);
+        return ResponseEntity.ok("The flight has been cancelled successfully.");
+    }
+
+    @PostMapping("/flights/update")
+    public ResponseEntity<String> updateFlight(@RequestBody FlightUpdateDTO flightDTO) {
+        flightManagmentService.updateFlight(flightDTO);
+        return ResponseEntity.ok("The flight has been updated successfully.");
+    }
```

Comment:
Using @PutMapping is more convenient because you are updating an existing flight.